### PR TITLE
Prevent device duplicates

### LINKED ISSUE #111: Devices list frequently incorrect
## Prerequisites
- [x] I have searched open and closed issues for duplicates.

## Describe the bug
The list of devices in the bluetooth indicator is freqently incorrect. 
Comparison of list in indicator with settings for example:
Screenshot from 2020-04-16 10-12-47@2x
Screenshot from 2020-04-16 18-39-03@2x

### To Reproduce
~~I'm afraid this is unclear. It happens over time.~~
1. Have some paired bluetooth deviced. 
2. Open indicator
3. Execute both or eather one of these commands:
    - sudo modprobe -r btusb && sudo modprobe btusb
    - sudo service bluetooth restart
4. Reopen indicator

I think it would be interesting to compare the logic in the indicator vs the settings and see if there are mistakes in the indicator. 

### Expected behavior
The list of devices in the indicator is consistent with the list of devices in the bluetooth settings. 

### Actual behavior
The list of devices in the indicator contains lots of duplicates. 

~~Note that even when keeping the bluetooth settings open there are no duplicates there.~~
Note that something similar happens in the bluetooth settings when executing restarting the bluetooth serve (`sudo service bluetooth restart`).

## Platform Information
Screenshot from 2020-04-16 10-18-05@2x

- [ ] I'm using the latest version from git that I've manually compiled
- [x] I'm using the latest released stable version

## Additional context
A first step was taken in this PR:
https://github.com/elementary/wingpanel-indicator-bluetooth/pull/17

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/91296653-devices-list-frequently-incorrect?utm_campaign=plugin&utm_content=tracker%2F59641246&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F59641246&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by peteruithoven**
Fixes #111

### CODE REVIEW

File: src/Widgets/PopoverWidget.vala [hunk 1]
```diff
@@ -92,16 +92,14 @@ public class BluetoothIndicator.Widgets.PopoverWidget : Gtk.Box {
 
 
         object_manager.device_added.connect ((device) => {
+            // Remove existing rows for this device
+            remove_device (device);
```
Comment (danrabbit) [inline src/Widgets/PopoverWidget.vala:96]: Seems kinda of silly to remove the device and then add it again. If the device row already exists, we should probably just return and do nothing

Comment (peteruithoven) [reply src/Widgets/PopoverWidget.vala:96]: But the old one isn't connected anymore. It at least didn't display the right state anymore when I tried such an approach. (It's also weird that the one that isn't connected anymore isn't removed)

Comment (danrabbit) [reply src/Widgets/PopoverWidget.vala:96]: Fair enough! Can you leave a comment so we know that this is done this way for a reason and someone doesn’t try to “clean up” later?

Comment (peteruithoven) [reply src/Widgets/PopoverWidget.vala:96]: Sure thing. I hope this is clear enough.

**Comment by peteruithoven** [comment]
Thanks for the review @danrabbit.

**Comment by danrabbit** [comment]
Thanks for your fix!

**Pull request merged**